### PR TITLE
Add the OAuth2 access token binding table.

### PR DESCRIPTION
**Pull request by thanujalk**
### Proposed changes in this pull request

- Add the OAuth2 access token binding table.

### When should this PR be merged

- NA

### Follow up actions

- NA

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql [hunk 1]
```diff
@@ -93,10 +93,21 @@ CREATE TABLE IDN_OAUTH2_ACCESS_TOKEN (
...
             IDP_ID INTEGER NOT NULL,
+            TOKEN_BINDING_REF VARCHAR (32) DEFAULT 'NONE' NOT NULL,
             PRIMARY KEY (TOKEN_ID),
             FOREIGN KEY (CONSUMER_KEY_ID) REFERENCES IDN_OAUTH_CONSUMER_APPS(ID) ON DELETE CASCADE,
             CONSTRAINT CON_APP_KEY UNIQUE (CONSUMER_KEY_ID,AUTHZ_USER,TENANT_ID,USER_DOMAIN,USER_TYPE,TOKEN_SCOPE_HASH,
-                                 TOKEN_STATE,TOKEN_STATE_ID,IDP_ID)) INDEX IN TS32K
+                                 TOKEN_STATE,TOKEN_STATE_ID,IDP_ID,TOKEN_BINDING_REF)) INDEX IN TS32K
+/
+
+CREATE TABLE IDN_OAUTH2_ACCESS_TOKEN_BINDING (
```
Comment (madurangasiriwardena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql]: Some databases have a limitation of 30 bytes for table names.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql [hunk 2]
```diff
@@ -93,10 +93,21 @@ CREATE TABLE IDN_OAUTH2_ACCESS_TOKEN (
...
+            TOKEN_ID VARCHAR (255) NOT NULL,
+            TOKEN_BINDING_TYPE VARCHAR (32) NOT NULL,
+            TOKEN_BINDING_REF VARCHAR (32) NOT NULL,
+            TOKEN_BINDING_VALUE VARCHAR (1024) NOT NULL,
+            TENANT_ID INTEGER DEFAULT -1,
```
Comment (madurangasiriwardena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql]: Do we need a default value for TENANT_ID?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql (hunk 1 above)
Comment (thanujalk) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql]: Done

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql (hunk 2 above)
Comment (thanujalk) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/db2.sql]: Other tables follow the same approach.

**Pull request merged**